### PR TITLE
ADD: No Network Connection alert

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by marcosrdz** [comment]
Screen Shot 2021-02-22 at 9 51 40 PM

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/hjk2491a3a8k6wfrf7876q5m24

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614049894.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2tf99ynvzfpr3vz0futufbbmqm

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614051406.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/1tnd26h4vyhbznwek8x94z9fyr

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614099753.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zc4v08y9berrytp9grnhtcd83c

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614118148.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dahygbfp9p11wtb71zymmewja4

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614126973.apk)

File: BlueComponents.js [hunk 1]
```diff
@@ -59,14 +60,35 @@ Platform.OS === 'android' ? (ActivityIndicator.defaultProps.color = PlatformColo
...
 export const BlueButton = props => {
   const { colors } = useTheme();
+  const { isConnected, isInternetReachable } = useNetInfo();
 
   let backgroundColor = props.backgroundColor ? props.backgroundColor : colors.mainColor || BlueCurrentTheme.colors.mainColor;
   let fontColor = props.buttonTextColor || colors.buttonTextColor;
-  if (props.disabled === true) {
+  const isButtonDisabled =
+    props.disabled === true ||
+    (props.disableOnNoNetworkAvailable === true && isConnected === false) ||
```
Comment (Overtorment) [inline BlueComponents.js:69]: this is overengineered.
whats the difference between isConnected and isInternetReachable ?
what difference does it make if internet is not reachable, but isConnected - user still cant do what he wanted so he doesn't give a shit if it isConnected

File: BlueComponents.js [hunk 2]
```diff
@@ -83,6 +105,11 @@ export const BlueButton = props => {
         paddingHorizontal: 16,
       }}
       {...props}
+      onPress={
+        (props.networkMustBeReachable && isConnected === false) || (props.internetMustBeReachable && isInternetReachable === false)
```
Comment (Overtorment) [inline BlueComponents.js:109]: overengineered. plus `internetMustBeReachable` is not even used

File: loc/en.json [hunk 3]
```diff
@@ -18,7 +19,9 @@
     "invalid_animated_qr_code_fragment" : "Invalid animated QRCode fragment. Please try again.",
     "file_saved": "File ({filePath}) has been saved in your Downloads folder.",
     "discard_changes": "Discard changes?",
-    "discard_changes_detail": "You have unsaved changes. Are you sure to discard them and leave the screen?"
+    "discard_changes_detail": "You have unsaved changes. Are you sure to discard them and leave the screen?",
+    "no_network_available_title": "Network Connection",
+    "no_network_available_description": "A network connection was not found. In order to proceed please verify your connection settings."
```
Comment (Overtorment) [inline loc/en.json:24]: maybe use less formal language?

Comment (Overtorment) [reply loc/en.json:24]: @ncoelho

File: BlueComponents.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ import loc, { formatBalance, formatBalanceWithoutSuffix, formatBalancePlain, rem
 import Lnurl from './class/lnurl';
 import { BlueStorageContext } from './blue_modules/storage-context';
 import ToolTipMenu from './components/TooltipMenu';
+import { useNetInfo } from '@react-native-community/netinfo';
```
Comment (Overtorment) [inline BlueComponents.js:45]: shit another dep...

**Review by Overtorment — CHANGES_REQUESTED** [review]

File: BlueComponents.js (hunk 1 above)
Comment (marcosrdz) [reply BlueComponents.js:69]: isConnected means that they are connected to a network with no internet access but can communicate to a node that might have internet. IsInternetReachable means that the user has to have Internet access

File: BlueComponents.js (hunk 2 above)
Comment (marcosrdz) [reply BlueComponents.js:109]: Its there for a scenario in which its required

File: BlueComponents.js (hunk 4 above)
Comment (marcosrdz) [reply BlueComponents.js:45]: Thats the nature of React Native. Everything is separate. But this is official from the RN community. Not some unknown third party

File: BlueComponents.js (hunk 1 above)
Comment (marcosrdz) [reply BlueComponents.js:69]: For example...lets say u have a device which is blocked from accessing the Internet, but is able to access the local network (your node).

**Review by limpbrains — APPROVED** [review]
I agree that `internetMustBeReachable` can be removed since it is unlikely that it will be needed

File: BlueComponents.js (hunk 1 above)
Comment (marcosrdz) [reply BlueComponents.js]: Removed

**Comment by marcosrdz** [comment]
> I agree that `internetMustBeReachable` can be removed since it is unlikely that it will be needed

Removed.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/w3k141pzk6j26qk7wa00ftzf7r

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614244982.apk)

File: loc/en.json (hunk 3 above)
Comment (ncoelho) [reply loc/en.json:24]: Looks ok to me. Just make sure we need another string for "network connection". There's already 2 of them, a "network" and "network error". Probably we can reuse "network error". Let's be careful on adding strings.

Comment (marcosrdz) [reply loc/en.json:24]: Its not really a "network error". Its the lack of network. No processing has been attempted for it
To be fall under error

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/kmv7r2uax91vtca1j30z1evt04

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614291234.apk)

File: loc/en.json (hunk 3 above)
Comment (marcosrdz) [reply loc/en.json:23]: Changed to loc.errors.network

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9400n6bwf720b9uku72taru894

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614366017.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/k75t278dadyma1d3x0hdu7yza0

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614443719.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/k2nu0b6euryz3vzytj1g0a02mg

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1614466157.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/j4f3cmxyvmqz2jqa6a2mw4jjgr

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1615385673.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/khmkh8aqbtnqp0gkm93yp8wzyc

[download apk](http://filestorage.bluewallet.io:1488/netinfo-1615388426.apk)

**Review by ncoelho — APPROVED** [review]

**Comment by marcosrdz** [comment]
@Overtorment

**Comment by ncoelho** [comment]
conflits

**Comment by Overtorment** [comment]
at this point i think this is overengineered, brings too little on the table, and adds complexity i don't want to deal with.

**Pull request closed without merging**